eclipse-collections-code-generator/src/main/resources/impl/map/mutable/primitiveObjectHashMap.stg

```diff
@@ -1652,6 +1652,35 @@ public class <name>ObjectHashMap\<V> implements Mutable<name>ObjectMap\<V>, Exte
         return null;
     }
 
+    public V getOnly()
+    {
+        if (1 \< this.size())
+        {
+            throw new IllegalStateException("Iterable has multiple values");
+        }
+
+        if (this.sentinelValues != null)
+        {
+            if (this.sentinelValues.containsZeroKey)
+            {
+                return this.sentinelValues.zeroValue;
+            }
+            if (this.sentinelValues.containsOneKey)
+            {
+                return this.sentinelValues.oneValue;
+            }
+        }
+        for (int i = 0; i \< this.keys.length; i++)
+        {
+            if (isNonSentinel(this.keys[i]))
+            {
+                return this.values[i];
+            }
+        }
+
+        throw new IllegalStateException("Iterable is empty.");
+    }
```

Comment:
If throwing different exceptions for the cases of empty and multiple values, then both checks should be at the beginning of the method. Otherwise we can spend O(n) time checking for the empty case.